src/models/TechRecordsDAO.ts

```diff
@@ -35,6 +35,29 @@ class TechRecordsDAO {
     }
   }
 
+  private readonly CriteriaIndexMap: {[key: string]: string} = {
+    systemNumber: "SysNumIndex",
+    partialVin: "PartialVinIndex",
+    vrm: "VRMIndex",
+    vin: "VinIndex",
+    trailerId: "TrailerIdIndex"
+  };
+
+  private queryBuilder(searchTerm: string, searchCriteria: SEARCHCRITERIA, query: QueryInput) {
+
+    Object.assign(query.ExpressionAttributeNames!, {
+      [`#${searchCriteria}`]: searchCriteria === SEARCHCRITERIA.VRM ? "primaryVrm" : searchCriteria,
+    });
```

Comment:
I tried a refactor with the `satisfies` operator to avoid a `!` but I think our typescript version doesn't have it yet 😢

Reply:
sad meep